src/inventory/stock-receipt/stock-receipt.jsx

```diff
@@ -1,516 +1,533 @@
-                                  {...getHeaderProps({
-                                    header,
-                                  })}
-                                  className={header.key === 'item' ? styles.stickyColumn : ''}
-                                >
-                                  {header.header}
-                                </TableHeader>
-                              ))}
-                            </TableRow>
-                          </TableHead>
-                          <TableBody>
-                            {rows.map((row) => (
-                              <TableRow {...getRowProps({ row })}>
-                                {row.cells.map((cell) => {
-                                  if (
-                                    cell.id.includes('totalQuantity') ||
-                                    cell.id.includes('quantity')
-                                  ) {
-                                    return (
-                                      <TableCell key={cell.id}>
-                                        <TextInput
-                                          size="sm"
-                                          id={cell.id}
-                                          value={cell.value}
-                                          invalid={isNaN(cell.value)}
-                                          invalidText="Please enter a valid number"
-                                          labelText={''}
-                                          onChange={(e) =>
-                                            updateActualQuantity(e.target.value, row, cell.id)
-                                          }
-                                        />
-                                      </TableCell>
-                                    );
-                                  } else
-                                    return (
-                                      <TableCell
-                                        key={cell.id}
-                                        className={
-                                          cell.id.includes('item') ? styles.stickyColumn : ''
-                                        }
-                                      >
-                                        {cell.value}
-                                      </TableCell>
-                                    );
-                                })}
-                              </TableRow>
-                            ))}
-                          </TableBody>
-                        </Table>
-                      </>
-                    )}
-                  </DataTable>
-                </div>
-              </Column>
-            )
-          )}
+	const handleInputChange = (id, field, value) => {
+		field === 'totalQuantity' && value < 0 ? setNegativeError(true) : setNegativeError(false);
+		setRows((prevRows) =>
+			prevRows.map((row) => (row.id === id ? { ...row, [field]: value } : row))
+		);
+	};
```

Comment:
We may remove these empty spaces in between code before merging

Reply:
done